core/src/main/java/io/grpc/internal/CallCredentialsApplyingTransportFactory.java

```diff
@@ -86,8 +87,12 @@ public ClientStream newStream(
         if (callOptions.getAuthority() != null) {
           effectiveAttrsBuilder.set(CallCredentials.ATTR_AUTHORITY, callOptions.getAuthority());
         }
-        creds.applyRequestMetadata(method, effectiveAttrsBuilder.build(),
-            firstNonNull(callOptions.getExecutor(), appExecutor), applier);
+        try {
+          creds.applyRequestMetadata(method, effectiveAttrsBuilder.build(),
+              firstNonNull(callOptions.getExecutor(), appExecutor), applier);
```

Comment:
applyRequestMetadata() is not supposed to throw. Errors should be passed to the applier.

Reply:
Since this is user-provided code, I don't feel like that negates the need to handle exceptions from the call. It's too easy for a NPE or similar to leak through. I've adding a message to the status to make it clearer the code did something wrong.